### DISCUSSION TITLE
Adding extra kernel modules

### DISCUSSION THREAD

**Discussion by Richzendy** [krglaws/MyLFS#21] [General]
How i can add extra modules to kernel?

I'm trying for example:

```
CONFIGFILE=config-$KERNELVERS
cat > config-extras << EOF
CONFIG_FUSE_FS=y
CONFIG_OVERLAY_FS=y
CONFIG_BRIDGE=y
CONFIG_NETDEVICES=y
CONFIG_NET_CORE=y
CONFIG_VETH=y
CONFIG_NETFILTER=y
CONFIG_NETFILTER_XTABLES=y
CONFIG_NETFILTER_XT_TARGET_LOG=y
EOF
make mrproper

if [ -f /boot/$CONFIGFILE ]
then
    cp /boot/$CONFIGFILE ./.config

    # Extra kernel modules added
    cat config-extras >> ./.config

    make olddefconfig
else
    make defconfig
fi
```
Into linux.sh script at phase4, but I  don't know if is the best approach

**Comment by krglaws** [comment]
First of all, you need to create a kernel config separately before running mylfs.sh. Unpack the linux tarball, and use the commands outlined in the LFS chapter where you build the kernel to generate your kernel config (this can vary depending on the BLFS packages you intend to add later on). Then when you run the script you can use the `-k path/to/config` command line option. As for getting the kernel to build specific drivers, here is a reddit post I found that seems to give some suggestions: [https://www.reddit.com/r/linuxquestions/comments/w9vyyv/how_do_i_get_my_kernel_build_to_include_a/](https://www.reddit.com/r/linuxquestions/comments/w9vyyv/how_do_i_get_my_kernel_build_to_include_a/)
